### DISCUSSION TITLE
Rejected non-opensource software because of "<30 forks <30 watchers <75 stars"

### DISCUSSION THREAD

**Discussion by golopupinsky** [Homebrew/discussions#1805] [Casks]
Hi everyone! 
I'd like to add a non-open source application that is distributed via github releases page to brew cask.
Automatic test firing on PR sees github in url, checks repo's forks/likes/stars and fails because obviously there's not that much likes and forks on the releases page. A person comes in later and closes [my PR](https://github.com/Homebrew/homebrew-cask/pull/108295) because "not notable enough". 

Effectively this means that no closed source software is allowed to host releases on github for cask to accept it.
Seems a little counter-intuitive to me. Is it intentional?

**Comment by SMillerDev** [comment]
> Effectively this means that no closed source software is allowed to host releases on github for cask to accept it.
Seems a little counter-intuitive to me. Is it intentional?

It seems very counter intuitive to me to host only compiled binaries on a source code management website. Are you sure that is following the github terms of service?

**Comment by golopupinsky** [comment]
Pretty sure they have a private github repository

**Reply by nandahkrishna** [reply]
The repository you used for the stable URL isn't private: https://github.com/conduktor/builds/. Private repositories' star count can't be retrieved through the API without necessary permissions (which we don't have). This is the only GitHub repository Homebrew knows about, and so it cares about that statistics for that repository.

**Reply by golopupinsky** [reply]
Yes, and here's why I ask my question. Imagine this company hosting their binaries somewhere else. Would my PR be rejected in that case? If not, there's clearly some flaw in review logic as hosting builds on github shouldn't reduce the chances for brew/cask approval. It should work vice versa.

**Reply by SMillerDev** [reply]
>  Imagine this company hosting their binaries somewhere else. Would my PR be rejected in that case? If not, there's clearly some flaw in review logic as hosting builds on github shouldn't reduce the chances for brew/cask approval.

The assumption is that if you use a collaboration platform like GitHub, your tools are likely open-source and can therefore be checked using common open-source popularity metrics.

If you host it on your own website we can't really check those metrics. If you can find a good source on app popularity I'd be willing to write an audit for self-hosted apps too.

**Reply by golopupinsky** [reply]
The assumption for open source software is clear and fair. 
For proprietary though some other approaches might be more effective. Like manual review. For example, if you study this company's github account you'll find it has other repos with dozens of likes/forks/watches ([1](https://github.com/conduktor/kafka-security-manager),[2](https://github.com/conduktor/kafka-stack-docker-compose)) which probably gives them some credibility.

In any case the fact that builds are available for download on github should not be the reason for semi-automatic PR rejection.

**Reply by SMillerDev** [reply]
So we're in agreement that the current method is the right one then. If it's on an open source hosting platform we check by open source source metrics. If it's proprietary we do a manual review.

**Reply by golopupinsky** [reply]
Open source software -> open source metrics
Proprietary software    -> manual review

Hosting platform is irrelevant.

**Reply by SMillerDev** [reply]
> Hosting platform is irrelevant.

I disagree, if it walks like a duck and quacks like a duck but it's really a genetically modified horse abusing a ducks body...I don't think it's Homebrew's fault for assuming it's a duck.

**Reply by golopupinsky** [reply]
Well, it IS a fault. A fault that Homebrew could have prevented. With not much effort and to mutual benefit. 
Deriving decision about source openness from build's host name is controversary to say the least. 

And it's not "abusing", it's a widespread practice for proprietary software to use private github source and public build repos.

Anyway, if you don't want to modify your policies can you at least re-review my PR? Since we more or less agreed that proprietary software shouldn't follow the 30/30/75 rule and there were no other issues with it.
